quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/StoragePropertyMappers.java

```diff
@@ -30,10 +31,161 @@ private StoragePropertyMappers(){}
+                        .build(),
+                fromOption(StorageOptions.STORAGE_CACHE_REALM_ENABLED)
+                        .to("kc.spi-realm-cache-default-enabled")
+                        .mapFrom("storage")
+                        .transformer(StoragePropertyMappers::isCacheAreaEnabledForStorage)
+                        .paramLabel(Boolean.TRUE + "|" + Boolean.FALSE)
+                        .build(),
+                fromOption(StorageOptions.STORAGE_CACHE_AUTHORIZATION_ENABLED)
+                        .to("kc.spi-authorization-cache-default-enabled")
+                        .mapFrom("storage")
+                        .transformer(StoragePropertyMappers::isCacheAreaEnabledForStorage)
+                        .paramLabel(Boolean.TRUE + "|" + Boolean.FALSE)
+                        .build(),
+                fromOption(StorageOptions.STORAGE_CACHE_USER_ENABLED)
+                        .to("kc.spi-user-cache-default-enabled")
+                        .mapFrom("storage")
+                        .transformer(StoragePropertyMappers::isCacheAreaEnabledForStorage)
+                        .paramLabel(Boolean.TRUE + "|" + Boolean.FALSE)
+                        .build(),
+                fromOption(StorageOptions.STORAGE_SINGLE_USE_OBJECT)
+                        .to("kc.spi-single-use-object-provider")
+                        .mapFrom("storage")
+                        .transformer(StoragePropertyMappers::getCacheStorage)
+                        .paramLabel("type")
+                        .build(),
         };
     }
+
+    private static Optional<String> getAreaStorage(Optional<String> storage, ConfigSourceInterceptorContext context) {
+        return of("legacy".equals(storage.orElse(null)) ? "jpa" : "map");
+    }
+
+    private static Optional<String> getCacheStorage(Optional<String> storage, ConfigSourceInterceptorContext context) {
+        return of("legacy".equals(storage.orElse(null)) ? "infinispan" : "map");
+    }
+
+    private static Optional<String> getDbLockProvider(Optional<String> storage, ConfigSourceInterceptorContext context) {
+        return of("legacy".equals(storage.orElse(null)) ? "jpa" : "none");
+    }
+
+    private static Optional<String> getUserSessionPersisterStorage(Optional<String> storage, ConfigSourceInterceptorContext context) {
+        return of("legacy".equals(storage.orElse(null)) ? "jpa" : "disabled");
+    }
+
+    private static Optional<String> isDefaultPersistenceUnitEnabled(Optional<String> value, ConfigSourceInterceptorContext context) {
+        if (value.get().equals(StorageOptions.StorageType.legacy.name())) {
+            return of(Boolean.TRUE.toString());
+        }
+
+        return of(Boolean.valueOf(value.get()).toString());
+    }
+
+    private static Optional<String> resolveStorageProvider(Optional<String> value, ConfigSourceInterceptorContext context) {
+        return Optional.ofNullable("legacy".equals(value.orElse(null)) ? null : "concurrenthashmap");
+    }
```

Comment:
I see that this resolve to `null` for the legacy storage provider, and to `concurrenthashmap` for everything else.

As we only want to support CHM, this is find for this PR. 

For the future, we would have for example "jpa" or "hotrod" here, but also "null" for any mixed storage. At least with the non-Quarkus runtimes I fell into the trap several times that I needed to set "null" here to have "jpa" running for some stores, and "concurrenthashmap" for others.

Whenever (in the old world) the "provider" was set, the lookup of any other provider was no longer working. 

If this will change in the next PR soonish, let's keep it as is.

Reply:
Yeah, that is correct in the sense that it will change in the future. As I mentioned before, most of those "internal" options should be temporary while we don't start the next round of changes to integrate jpa, "mixed", and even the tree store.

I did not have much experience on running multiple stores so I can't yet give you much about this topic :)

But yeah, one step at a time.